### PR TITLE
Refactor CodeActSWEAgent, add response parser

### PR DESCRIPTION
**Pull request by yufansong**
Find we forget to add integration test for `CodeActSWEAgent`. Ready for review.

### CODE REVIEW

**Comment by tobitege** [comment]
Might just be a personal preference, but would it be cleaner for the ResponseParser and ActionParser being in separate files?

**Comment by yufansong** [comment]
> Might just be a personal preference, but would it be cleaner for the ResponseParser and ActionParser being in separate files?

Sure, not problem. Will do it later.

File: agenthub/codeact_swe_agent/response_parser.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    def __init__(
+        self,
+    ):
+        # Need pay attention to the item order in self.action_parsers
+        self.action_parsers = [
+            CodeActSWEActionParserFinish(),
+            CodeActSWEActionParserCmdRun(),
+            CodeActSWEActionParserIPythonRunCell(),
+            CodeActSWEActionParserBrowseInteractive(),
```
Comment (li-boxuan) [inline agenthub/codeact_swe_agent/response_parser.py:30]: I think we could remove this now?

**Review by li-boxuan — APPROVED** [review]

File: agenthub/codeact_swe_agent/action_parser.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import re
+
+from opendevin.controller.action_parser import ActionParser
+from opendevin.events.action import (
+    Action,
+    AgentFinishAction,
+    BrowseInteractiveAction,
```
Comment (li-boxuan) [inline agenthub/codeact_swe_agent/action_parser.py:7]:
```suggestion
```

File: agenthub/codeact_swe_agent/action_parser.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        code_group = self.python_code.group(1).strip()
+        thought = action_str.replace(self.python_code.group(0), '').strip()
+        return IPythonRunCellAction(
+            code=code_group,
+            thought=thought,
+            kernel_init_code=self.jupyter_kernel_init_code,
+        )
+
+
+class CodeActSWEActionParserBrowseInteractive(ActionParser):
```
Comment (li-boxuan) [inline agenthub/codeact_swe_agent/action_parser.py:98]: could be removed

File: agenthub/codeact_swe_agent/response_parser.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        ]
+        self.default_parser = CodeActSWEActionParserMessage()
+
+    def parse(self, response: str) -> Action:
+        action_str = self.parse_response(response)
+        return self.parse_action(action_str)
+
+    def parse_response(self, response) -> str:
+        action = response.choices[0].message.content
+        for lang in ['bash', 'ipython', 'browse']:
```
Comment (li-boxuan) [inline agenthub/codeact_swe_agent/response_parser.py:40]:
```suggestion
        for lang in ['bash', 'ipython']:
```

File: agenthub/codeact_swe_agent/response_parser.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+from opendevin.controller.action_parser import ResponseParser
+from opendevin.events.action import Action
+
+
+class CodeActSWEResponseParser(ResponseParser):
+    """
+    Parser action:
+        - CmdRunAction(command) - bash command to run
+        - IPythonRunCellAction(code) - IPython code to run
+        - AgentDelegateAction(agent, inputs) - delegate action for (sub)task
```
Comment (li-boxuan) [inline agenthub/codeact_swe_agent/response_parser.py:17]: I don't think this agent delegates

```suggestion
```

File: agenthub/codeact_swe_agent/response_parser.py (hunk 1 above)
Comment (yufansong) [reply agenthub/codeact_swe_agent/response_parser.py:30]: I am little confuse. Why? `self.action_parsers` will used in `parse_action`

Comment (yufansong) [reply agenthub/codeact_swe_agent/response_parser.py:30]: Ah, get what you mean, You mean delete `CodeActSWEActionParserBrowseInteractive `

**Pull request merged**